rust/src/builders/tx_builder.rs

```diff
@@ -1276,25 +1045,13 @@ impl TransactionBuilder {
         }
     }
 
-    // This function should be producing the total witness-set
-    // that is created by the tx-builder itself,
-    // before the transaction is getting signed by the actual wallet.
-    // E.g. scripts or something else that has been used during the tx preparation
-    fn get_witness_set(&self) -> TransactionWitnessSet {
-        let mut wit = TransactionWitnessSet::new();
-        if let Some(scripts) = self.native_scripts.as_ref() {
-            wit.set_native_scripts(scripts);
-        }
-        wit
-    }
-
     /// Returns full Transaction object with the body and the auxiliary data
     /// NOTE: witness_set will contain all mint_scripts if any been added or set
     /// NOTE: is_valid set to true
     pub fn build_tx(&self) -> Result<Transaction, JsError> {
         Ok(Transaction {
             body: self.build()?,
-            witness_set: self.get_witness_set(),
+            witness_set: self.witness_set_builder.build()?,
```

Comment:
This won't work properly because this will also include all the useless fake witnesses we created purely for tx fee calculation

Reply:
In fact, this will include no fake witness because it would fail at remaining witnesses check. An error is returned instead. However, this prevents the fee being calculated correctly. I've extract the `build()` for fee calculation. Please review.